bin/nvcc_wrapper

```diff
@@ -242,6 +242,76 @@ do
     cuda_args="$cuda_args $1 $2"
     shift
     ;;
+  # Handle Werror. Note, we must differentiate between the ones going to nvcc and the host compiler
+  # --Werror kind,... OR --Werror=kind,... <- always to nvcc
+  --Werror)
+    cuda_args="$cuda_args $1 $2"
+    shift
+    ;;
+  --Werror=*)
+    cuda_args="$cuda_args $1"
+    ;;
+  # -Werror kind,... where kind is one of {all-warnings, cross-execution-space-call, reorder, default-stream-launch, missing-launch-bounds, ext-lambda-captures-this, deprecated-declarations} <- goes to nvcc
+  # -Werror not followed by any kind as mentioned above goes to host compiler without any arguments
+  -Werror)
+    if [ $# -gt 1 ]; then
+      kinds=(${2//,/ })
+      first_kind=${kinds[0]}
+      # check if the first kind is one of the allowed ones, then this must be an nvcc list so put all of them to the cuda compiler
+      case $first_kind in
+      all-warnings|cross-execution-space-call|reorder|default-stream-launch|missing-launch-bounds|ext-lambda-captures-this|deprecated-declarations)
+        cuda_args="$cuda_args $1 $2"
+        shift
+        ;;
+      *)
+        if [ $first_xcompiler_arg -eq 1 ]; then
+          xcompiler_args="$1"
+          first_xcompiler_arg=0
+        else
+          xcompiler_args="$xcompiler_args,$1"
+        fi
+        ;;
+      esac
+    fi
+    ;;
+  # -Werror=kind,... will be split into two parts, those kinds that belong to nvcc (see above) go there, while all others go towards the host compiler
+  -Werror=*)
+    kinds_str=(${1:8}) # strip -Werror=
+    kinds=(${kinds_str//,/ })
+    first_werror_cuda=1
+    first_werror_host=1
+    # loop over all kinds that are sparated via ','
```

Comment:
```suggestion
    first_werror_host=1
    xcompiler_args_werror=
    # loop over all kinds that are sparated via ','
```
Otherwise, we get
```bash
$ ../bin/nvcc_wrapper -Werror=format -Werror=restrict -Werror=deprecated-declarations,ext-lambda-captures-this --show dummy.cpp 
nvcc -Werror=deprecated-declarations,ext-lambda-captures-this -ccbin g++ -arch=sm_35 -Xcompiler -Werror=format,-Werror=format-Werror=restrict -x cu dummy.cpp
```
instead of
```bash
$ ../bin/nvcc_wrapper -Werror=format -Werror=restrict -Werror=deprecated-declarations,ext-lambda-captures-this --show dummy.cpp 
nvcc -Werror=deprecated-declarations,ext-lambda-captures-this -ccbin g++ -arch=sm_35 -Xcompiler -Werror=format,-Werror=restrict -x cu dummy.cpp
```